skimage/io/tests/test_collection.py

```diff
@@ -104,29 +103,25 @@ def set_files(f):
             set_files('newfiles')
 
     @pytest.mark.skipif(not has_pooch, reason="needs pooch to download data")
-    def test_custom_load_func_sequence(self):
-        filename = fetch('data/no_time_for_that_tiny.gif')
-
+    def test_custom_load_func_sequence(self, random_gif_path):
         def reader(frameno):
-            vid = imageio.get_reader(filename)
-            return vid.get_data(frameno)
+            return imread(random_gif_path, mode="RGB")[frameno, ...]
```

Comment:
```suggestion
            return imread(random_gif_path, index=frameno, mode="RGB")
```

No need to read all frames. Just ask ImageIO to return the index you want. It will `seek` as close to the frame as possible and decode only what it needs to build that one image.